### ISSUE TITLE
Documentation Suggestions

### ISSUE THREAD

**Issue by CaelanStewart** [area17/twill#1233]
## Summary

Hi,

I'm approaching Twill as a new user, with moderate Laravel experience.

I really like Twill so far, and I can see its potential to save time with the right toolkits and component libraries – however, there were a few things I felt were under-explained in the documentation. Things which I feel detract from the speed of onboarding without people around that already know it well to answer questions.

The issues I found were, fairly generally:

1. There are lots of tidbits that are quite important that relate to various concepts that are distributed around, and not clearly or explicitly stated up front in important information boxes or bullet points, such as whether migrations are needed for blocks and fields
2. Ambiguous language open to multiple interpretations, where I'm left wondering which interpretation is correct, having to try and narrow it down
3. It isn't clearly explained how to create a singular pages which are used, for example, as a front-page. It is sort of by example in the buckets section, but really a section on singular pages (or even a home page example under a specific homepage section in the docs)
4. It isn't clear how repeaters are rendered – everything about actually rendering the content of blocks and repeaters is left very, very open to interpretation. I know the point is that Twill is fairly unopinionated on how content _should_ be rendered on the frontend, but this makes it difficult to figure out from your docs alone the best ways of approaching this given the semantics of your framework
5. No use of alert boxes containing important information, such as how Laravel handles these things
6. It doesn't clearly and comprehensively describe the purpose, allowed types and values of arguments/options to many Twill blade functions and class methods

## Examples

**Example 1**
Under "Form fields > Files" you have this code snippet:

`public $filesParams = ['file_role', ...]; // a list of file roles`

The term 'role' is already used for User roles, and presents an alternative and misleading interpretation of the above comment, which had to be clarified.

**Example 2**
It isn't described very clearly where the icons come from, except as an unexplained option in the default twill.php configuration section. These things were fairly easy to find out, but I feel there have been a lot of these things, and together culminate in a consistent theme of lacklustre documentation.

## Describe the solution you'd like

1. Use more explicit language, and carefully consider possible misinterpretations, and proactively mitigate for them by making the language and terminology completely clear, unambiguous and comprehensive.

2. Use alert sections like seen on Laravel:

<img width="605" alt="Screenshot 2021-11-02 at 12 00 09" src="https://user-images.githubusercontent.com/11272013/139847282-5cf97eaa-dc48-48ee-9f13-f268d5a84894.png">
<img width="633" alt="Screenshot 2021-11-02 at 12 00 34" src="https://user-images.githubusercontent.com/11272013/139847296-d720e7f6-117e-4842-b439-93146eb67027.png">

3. Have more complete and thorough examples, like a complete demo project that uses every feature of Twill in some simple and common use case.

4. Have more detailed tables and descriptions of options and their allowed values and types. WordPress, which I'm very happy to move away from, was admittedly very good at documentation, same with PHP. They are more mature projects of course, but I'd love to see Twill become the replacement for WordPress!

**Comment by CaelanStewart** [comment]
Have updated the issue now (after accidentally posting it half-baked!)

**Comment by CaelanStewart** [comment]
I have found some more issues with the documentation.

The primary thing I think is leaky abstractions, there have been so many little gotchas to consider as you drill down into the details, it's getting a bit frustrating. This was not my experience with Laravel or even any other framework or library I have used.

- $mediaParams is hardly explained at all. I want to have an image be resized, but I don't want it to be cropped. It isn't clear how to do this, and there is no mention of this I can find in the docs.
- By default, using the local media drive, transparent PNGs are converted to JPGs with a white background (and it isn't clear how to make it just use the same format as the original without digging into Glide and the specifics of how Twill merges default options (passing null to 'fm' in 'default_params' in config/glide.php doesn't work) with those passed. If it uses the null coalescing operator and passes a string as its right operand then my `null` will have no effect, so I have to read all of your code to find this out).
- There is no documentation on getting modules by slug (except halfway through one of your tutorial videos (which I had to play at 2x speed because he talks too slowly)), or general best practice approaches of generating URLs to module instances and the best places to put this code.

I'll keep posting the things I find here, as despite the docs I do for the most part really like Twill.
